src/RedHat.AspNetCore.Server.Kestrel.Transport.Linux/TransportThread.ThreadContext.cs

```diff
@@ -111,6 +112,38 @@ public unsafe ThreadContext(TransportThread transportThread)
                     AioInterop.IoSetup(EventBufferLength, &ctx).ThrowOnError();
                     _aioContext = ctx;
                 }
+
+                // Single reader, single writer queue since all writes happen from the TransportThread and reads happen sequentially
+                _acceptQueue = Channel.CreateUnbounded<TSocket>(new UnboundedChannelOptions
+                {
+                    SingleReader = true,
+                    SingleWriter = true,
+                    AllowSynchronousContinuations = _transportOptions.ApplicationSchedulingMode == PipeScheduler.Inline,
+                });
```

Comment:
I don't like that this queue is unbounded. There is nothing limiting the nr of sockets we're accepting.
If there are to many sockets queued, we shouldn't accept more.
You don't need to fix it, maybe add a comment.